tests/test_client.py

```diff
@@ -69,3 +75,198 @@ async def test_abstract_auth_sse(
     assert request
     assert request.headers["authorization"] == f"Bearer {TEST_ACCESS_TOKEN}"
     assert request.url.query.decode(encoding="utf-8") == "key1=value1&key2=value2"
+
+
+@pytest.mark.parametrize(
+    "event_data",
+    [
+        {
+            "items": [
+                {
+                    "handling": "none",
+                    "key": EventKey.DISHCARE_DISHWASHER_OPTION_HALF_LOAD.value,
+                    "level": "hint",
+                    "timestamp": 1733611453,
+                    "uri": "/a/random/relative/uri",
+                    "value": True,
+                },
+                {
+                    "handling": "none",
+                    "key": EventKey.BSH_COMMON_OPTION_REMAINING_PROGRAM_TIME.value,
+                    "level": "hint",
+                    "timestamp": 1733611453,
+                    "unit": "seconds",
+                    "uri": "/a/random/relative/uri",
+                    "value": 13800,
+                },
+            ]
+        },
+        {
+            "items": [
+                {
+                    "handling": "acknowledge",
+                    "key": EventKey.BSH_COMMON_EVENT_PROGRAM_ABORTED.value,
+                    "level": "hint",
+                    "timestamp": 1733616930,
+                    "value": "BSH.Common.EnumType.EventPresentState.Present",
+                }
+            ],
+        },
+        {
+            "handling": "none",
+            "key": "BSH.Common.Appliance.Disconnected",
+            "level": "hint",
+            "timestamp": 1733611817,
+            "value": True,
+        },
+        None,
+    ],
+)
+async def test_stream_all_events(
+    httpx_client: AsyncClient, httpx_mock: HTTPXMock, event_data: dict[str, Any] | None
+) -> None:
+    """Test the abstract auth sse."""
+    ha_id = "SIEMENS-HCS02DWH1-6BE58C26DCC1"
+    event_type = EventType.NOTIFY
+    httpx_mock.add_response(
+        url="https://example.com/api/homeappliances/events",
+        stream=IteratorStream(
+            [
+                "\n".join(
+                    [
+                        f"id: {ha_id}",
+                        f"data: {json.dumps(event_data) if event_data else ""}",
+                        f"event: {event_type}",
+                        "\n",
+                    ]
+                ).encode()
+            ]
+        ),
+        headers={"Content-Type": "text/event-stream"},
+    )
+
+    client = Client(AuthClient(httpx_client, "https://example.com"))
+
+    async for event_message in client.stream_all_events():
+        assert event_message.ha_id == ha_id
+        assert event_message.type == event_type
+        if not event_data:
```

Comment:
Add the expected list of `event_message`as a test parameter instead. Then we can just call next on that here and assert that it's equal to the event message from the stream.